File: apps/user/src/model/connection/ConnectionRequest.ts
'use strict'

import {RequestStatus} from "./RequestStatus";
import {DEFAULT_ALL_PERMISSION_GROUP_NAME} from "../Constants";
class ConnectionRequest {
    requestingUserId: string;
    requestedUserId: string;
    requesterFirstName: string;
    requesterLastName: string;
    requesterImage: string;
    requestingPermissionGroupName: string;
    status: RequestStatus;
    createDate: string;
    createBy: string;
    updateDate: string;
    updateBy: string;
    constructor(body: object) {
        // @ts-ignore
        const {requestingUserId, requestedUserId, requesterFirstName, requesterLastName, requesterImage, requestingPermissionGroupName, status, createDate, updateDate, updateBy, createBy} = body;

        // data fields
        this.requestingUserId = requestingUserId
        this.requestedUserId = requestedUserId
        this.requesterFirstName = requesterFirstName
        this.requesterLastName = requesterLastName
        this.requestingPermissionGroupName = requestingPermissionGroupName || DEFAULT_ALL_PERMISSION_GROUP_NAME
        this.requesterImage = requesterImage || process.env.DEFAULT_PROFILE_IMAGE
        this.status = status

        // audit fields
        this.createDate = createDate
        this.createBy = createBy
        this.updateDate = updateDate
        this.updateBy = updateBy
    }
}

export {ConnectionRequest}

File: apps/user/src/model/connection/connection_request.py
import os
from dataclasses import dataclass

from src.model.constants import DEFAULT_ALL_PERMISSION_GROUP_NAME
from src.model.document import ConnectionRequestDocument
from src.model.enums import RequestStatus
from src.model.view import View


@dataclass
class ConnectionRequestView(View):
    requestingUserId: str
    requestedUserId: str
    requesterFirstName: str
    requesterLastName: str
    requesterImage: str
    requestingPermissionGroupName: str
    status: RequestStatus

    @staticmethod
    def from_doc(doc: ConnectionRequestDocument):
        return ConnectionRequestView(
            requestingUserId=doc.get("requestingUserId"),
            requestedUserId=doc.get("requestedUserId"),
            requesterFirstName=doc.get("requesterFirstName"),
            requesterLastName=doc.get("requesterLastName"),
            requesterImage=doc.get("requesterImage", os.environ["DEFAULT_PROFILE_IMAGE"]),
            requestingPermissionGroupName=doc.get("requestingPermissionGroupName", DEFAULT_ALL_PERMISSION_GROUP_NAME),
            status=doc.get("status")
        )
